src/utils/functions.ts

```diff
@@ -284,6 +287,114 @@ const generateShareTemplate = (
   return baseMessage;
 };
 
+const addPlaylist = async (playlistUrl: string) => {
+  const playlistId = extractPlaylistId(playlistUrl);
+
+  if (!playlistId) {
+    throw new Error('Invalid playlist URL');
+  }
+
+  const existingPlaylist = await checkPlaylistExistsByPlaylistId(playlistId);
+
+  if (existingPlaylist.exists) {
+    return existingPlaylist.message;
+  }
+
+  const metadata = await fetchPlaylistMetadata(playlistId);
+
+  if (!metadata || metadata.success === false) {
+    throw new Error('Failed to fetch playlist metadata from YouTube');
+  }
+
+  const allVideos = await fetchPlaylistVideos(playlistId);
+
+  const { success, message } = await addPlaylistToDB({
+    playlistId,
+    playlistName: metadata.playlistName,
+    description: metadata.description,
+    videos: allVideos,
+  });
+
+  if (!success) {
+    throw new Error(message || 'Failed to add playlist');
+  }
+
+  return message;
+};
+
+// Fetch metadata for a YouTube playlist using its ID.
+export const fetchPlaylistMetadata = async (playlistId: string) => {
+  try {
+    const metadataResponse: any = await fetch(
+      `${YOUTUBE_API_PATH}/playlists?part=snippet&id=${playlistId}&key=${process.env.YOUTUBE_API_KEY}`
+    );
+    const metadata = await metadataResponse.json();
+
+    if (!metadataResponse.ok) {
+      throw new Error(`Failed to fetch playlist metadata: ${metadata.error.message}`);
+    }
+
+    if (!metadata.items || metadata.items.length === 0) {
+      throw new Error('No playlist metadata found.');
+    }
+
+    const playlistName = metadata.items[0].snippet.title;
+    const description = metadata.items[0].snippet.description;
+
+    return { playlistName, description };
+  } catch (error) {
+    return {
```

Comment:
Util functions don't need to  return status or api like response. Utils are just our utility classes.